hazelcast/src/main/java/com/hazelcast/json/internal/JsonSchemaNameValue.java

```diff
@@ -106,21 +106,32 @@ public int hashCode() {
     @Override
     public String toString() {
         return "JsonSchemaNameValue{"
-            + "nameStart=" + nameStart
-            + ", value=" + value
-            + '}';
+                + "nameStart=" + nameStart
+                + ", value=" + value
+                + '}';
     }
 
     @Override
     public void writeData(ObjectDataOutput out) throws IOException {
         out.writeInt(nameStart);
-        out.writeObject(value);
+        if (value instanceof JsonSchemaTerminalNode) {
+            out.writeBoolean(true);
+        } else {
+            out.writeBoolean(false);
+        }
```

Comment:
To remove if-else, we can have something like this:
```suggestion
        out.writeBoolean(value instanceof JsonSchemaTerminalNode);
```